<!doctype html>
<html class="no-js" lang="en" data-content_root="">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />
<link rel="index" title="Index" href="genindex.html" /><link rel="search" title="Search" href="search.html" />

    <!-- Generated with Sphinx 6.0.0 and Furo 2024.01.29 -->
        <title>Human Cortex Example - GPS 1.60 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/styles/furo.css?digest=369552022d0b975c8e74270ce6eabe0fb7978f24" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx_charts/charts.css" />
    <link rel="stylesheet" type="text/css" href="_static/tabs.css" />
    <link rel="stylesheet" type="text/css" href="_static/styles/furo-extensions.css?digest=30d1aed668e5c3a91c3e3bf6a60b675221979f0e" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="index.html"><div class="brand">GPS 1.60 documentation</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="index.html">
  
  
  <span class="sidebar-brand-text">GPS 1.60 documentation</span>
  
</a><form class="sidebar-search-container" method="get" action="search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="install.html">Installation guide</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="tutorials.html">GPS Tutorials</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle navigation of GPS Tutorials</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="mouse.html">Mouse Embryo Example</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_format.html">Data Format</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="data.html">GPS Resources Download</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="api.html">GPS Command Line Alphabets</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" role="switch" type="checkbox"/><label for="toctree-checkbox-2"><div class="visually-hidden">Toggle navigation of GPS Command Line Alphabets</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="api/find_latent_representations.html">step 1: find_latent_representation</a></li>
<li class="toctree-l2"><a class="reference internal" href="api/latent_to_gene.html">Step 2: latent_to_gene</a></li>
<li class="toctree-l2"><a class="reference internal" href="api/generate_ldscore.html">Step 3: generate_ldscore</a></li>
<li class="toctree-l2"><a class="reference internal" href="api/spatial_ldsc.html">Step 4: spatial_ldsc</a></li>
<li class="toctree-l2"><a class="reference internal" href="api/cauchy_combination.html">Step 5: cauchy_combination (optional)</a></li>
<li class="toctree-l2"><a class="reference internal" href="api/visualization.html">Step 6: visualization</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="release.html">Release notes</a></li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          
<div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main">
          <section id="human-cortex-example">
<h1>Human Cortex Example<a class="headerlink" href="#human-cortex-example" title="Permalink to this heading">#</a></h1>
<section id="preparation">
<h2>1. Preparation<a class="headerlink" href="#preparation" title="Permalink to this heading">#</a></h2>
<p>First you should <a class="reference internal" href="install.html"><span class="doc std std-doc">install</span></a> the GPS package.</p>
<section id="download-running-dependencies">
<h3>1.1 Download running dependencies<a class="headerlink" href="#download-running-dependencies" title="Permalink to this heading">#</a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>http://cnsgenomics.com/data/GPS/GPS_running_dependencies.tar.gz
tar<span class="w"> </span>-xvzf<span class="w"> </span>GPS_running_dependencies.tar.gz
</pre></div>
</div>
</section>
<section id="download-example-data">
<h3>1.2 Download example data<a class="headerlink" href="#download-example-data" title="Permalink to this heading">#</a></h3>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>http://cnsgenomics.com/data/GPS/GPS_example_data.tar.gz
tar<span class="w"> </span>-xvzf<span class="w"> </span>GPS_example_data.tar.gz
</pre></div>
</div>
<p>Directory structure</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>tree<span class="w"> </span>-L<span class="w"> </span><span class="m">2</span>
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>example_data
├── GWAS
│   ├── GIANT_EUR_Height_2022_Nature.sumstats.gz
│   ├── gwas_config.yaml
│   ├── IQ_NG_2018.sumstats.gz
│   └── BCX2_MCHC_EA_GWAMA.sumstats.gz
└── ST
    ├── Cortex_151507.h5ad
    └── E16.5_E1S1.MOSTA.h5ad
GPS_resource
├── genome_annotation
│   ├── enhancer
│   └── gtf
├── homologs
│   ├── macaque_human_homologs.txt
│   └── mouse_human_homologs.txt
├── LD_Reference_Panel
│   └── 1000G_EUR_Phase3_plink
└── LDSC_resource
    ├── hapmap3_snps
    └── weights_hm3_no_hla
</pre></div>
</div>
</section>
</section>
<section id="run-gps">
<h2>2. Run GPS<a class="headerlink" href="#run-gps" title="Permalink to this heading">#</a></h2>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="c1"># Constants and configuration</span>
<span class="nv">WORKDIR</span><span class="o">=</span><span class="s1">&#39;./example/Human_Cortex&#39;</span><span class="w"> </span><span class="c1"># This should be the directory where the GPS output will be saved</span>
<span class="nv">SAMPLE_NAME</span><span class="o">=</span><span class="s2">&quot;Cortex_151507&quot;</span><span class="w"> </span><span class="c1"># This should be the name of the sample</span>

<span class="c1"># Input data</span>
<span class="nv">HDF5_PATH</span><span class="o">=</span><span class="s2">&quot;example_data/ST/Cortex_151507.h5ad&quot;</span>
<span class="nv">ANNOTATION</span><span class="o">=</span><span class="s2">&quot;layer_guess&quot;</span><span class="w"> </span><span class="c1"># This should be the annotation layer in the ST data</span>
<span class="nv">DATA_TYPE</span><span class="o">=</span><span class="s1">&#39;count&#39;</span><span class="w"> </span><span class="c1"># This should be the gene expression layer in the ST data, e.g. &#39;count&#39; or &#39;log1p&#39;</span>

<span class="c1"># Running Dependencies and Resources</span>
<span class="nv">GTFFILE</span><span class="o">=</span><span class="s2">&quot;GPS_resource/genome_annotation/gtf/gencode.v39lift37.annotation.gtf&quot;</span>
<span class="nv">BRAIN_ENHANCER_FILE</span><span class="o">=</span><span class="s2">&quot;GPS_resource/genome_annotation/enhancer/by_tissue/BRN/ABC_roadmap_merged.bed&quot;</span>
<span class="nv">BFILE_ROOT</span><span class="o">=</span><span class="s2">&quot;GPS_resource/LD_Reference_Panel/1000G_EUR_Phase3_plink/1000G.EUR.QC&quot;</span>
<span class="nv">KEEP_SNP_ROOT</span><span class="o">=</span><span class="s2">&quot;GPS_resource/LDSC_resource/hapmap3_snps/hm&quot;</span>
<span class="nv">W_FILE</span><span class="o">=</span><span class="s2">&quot;GPS_resource/LDSC_resource/weights_hm3_no_hla/weights.&quot;</span>
</pre></div>
</div>
<p>After setting up the constants and configuration, execute the following steps in order to run the GPS analysis on the human cortex data:</p>
<section id="find-latent-representations">
<h3>2.1 find_latent_representations<a class="headerlink" href="#find-latent-representations" title="Permalink to this heading">#</a></h3>
<p><strong>Objective</strong>: This initial step is for addressing technical noise and sparsity in spatial transcriptomics data. By employing the GNN model, the GPS finds latent representations for each spot.</p>
<p><strong>Input</strong>:</p>
<ul class="simple">
<li><p>HDF5 file containing ST data (<code class="docutils literal notranslate"><span class="pre">HDF5_PATH</span></code>) with cell type annotation (<code class="docutils literal notranslate"><span class="pre">ANNOTATION</span></code>).</p></li>
</ul>
<p><strong>Output</strong>: A .h5ad file containing the ST data with the refined latent representations for each spot.</p>
<p><strong>Execution</strong>:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nv">HDF5_WITH_LATENT_PATH</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$WORKDIR</span><span class="s2">/</span><span class="nv">$SAMPLE_NAME</span><span class="s2">/find_latent_representations/</span><span class="si">${</span><span class="nv">SAMPLE_NAME</span><span class="si">}</span><span class="s2">_add_latent.h5ad&quot;</span>
GPS<span class="w"> </span>run_find_latent_representations<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--input_hdf5_path<span class="w"> </span><span class="nv">$HDF5_PATH</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--sample_name<span class="w"> </span><span class="nv">$SAMPLE_NAME</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--output_hdf5_path<span class="w"> </span><span class="nv">$HDF5_OUTPUT</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--annotation<span class="w"> </span><span class="nv">$ANNOTATION</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--type<span class="w"> </span><span class="nv">$DATA_TYPE</span>
</pre></div>
</div>
</section>
<section id="latent-to-gene">
<h3>2.2 latent_to_gene<a class="headerlink" href="#latent-to-gene" title="Permalink to this heading">#</a></h3>
<p><strong>Objective</strong>: Building on the latent representations, this step leverages the latent representations of the spot and its neighbors to generate gene marker scores.</p>
<p><strong>Input</strong>:</p>
<ul class="simple">
<li><p>HDF5 file with latent representations from the previous step (<code class="docutils literal notranslate"><span class="pre">HDF5_WITH_LATENT_PATH</span></code>).</p></li>
</ul>
<p><strong>Output</strong>:</p>
<ul class="simple">
<li><p>A .feather file with gene marker scores (<code class="docutils literal notranslate"><span class="pre">MKSCORE_FEATHER_PATH</span></code>).</p></li>
</ul>
<p><strong>Execution</strong>:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nv">MKSCORE_FEATHER_PATH</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$WORKDIR</span><span class="s2">/</span><span class="nv">$SAMPLE_NAME</span><span class="s2">/latent_to_gene/</span><span class="si">${</span><span class="nv">SAMPLE_NAME</span><span class="si">}</span><span class="s2">_gene_marker_score.feather&quot;</span>
GPS<span class="w"> </span>run_latent_to_gene<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--input_hdf5_with_latent_path<span class="w"> </span><span class="nv">$HDF5_WITH_LATENT_PATH</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--sample_name<span class="w"> </span><span class="nv">$SAMPLE_NAME</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--output_feather_path<span class="w"> </span><span class="nv">$MKSCORE_FEATHER_PATH</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--latent_representation<span class="w"> </span><span class="s2">&quot;latent_GVAE&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--num_neighbour<span class="w"> </span><span class="m">51</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--num_neighbour_spatial<span class="w"> </span><span class="m">201</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--annotation<span class="w"> </span><span class="nv">$ANNOTATION</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--type<span class="w"> </span><span class="nv">$DATA_TYPE</span>
</pre></div>
</div>
</section>
<section id="generate-ldscore">
<h3>2.3 generate_ldscore<a class="headerlink" href="#generate-ldscore" title="Permalink to this heading">#</a></h3>
<p><strong>Objective</strong>: By using the gene marker scores from the previous step, GPS first assigns gene specificity scores to each SNP by referencing the gene annotation data. Specifically, GPS assigns gene specificity scores by evaluating each SNP’s closeness to the gene transcription start sites (TSS) and, optionally, enhancer-gene linking maps. Then use the gene specificity scores to generates LD scores for each SNP.</p>
<p><strong>Input</strong>:</p>
<ul class="simple">
<li><p>Feather file with gene marker scores from the previous step (<code class="docutils literal notranslate"><span class="pre">MKSCORE_FEATHER_FILE</span></code>).</p></li>
<li><p>Reference panel data (<code class="docutils literal notranslate"><span class="pre">BFILE_ROOT</span></code> and <code class="docutils literal notranslate"><span class="pre">KEEP_SNP_ROOT</span></code>).</p></li>
<li><p>Gene annotation file.</p></li>
</ul>
<p><strong>Output</strong>:</p>
<ul class="simple">
<li><p>A set of LD score chunks stored in the (<code class="docutils literal notranslate"><span class="pre">LDScoreDir</span></code>).</p></li>
</ul>
<p><strong>Three SNP to gene linking methods are available:</strong></p>
<div class="tab-set docutils">
<input checked="True" class="tab-input" id="tab-set--0-input--1" name="tab-set--0" type="radio"><label class="tab-label" for="tab-set--0-input--1">1. Use TSS Only</label><div class="tab-content docutils">
<p>This will use TSS only to link SNPs to gene specificity.</p>
<p><code class="docutils literal notranslate"><span class="pre">--gene_window_size</span> <span class="pre">=</span> <span class="pre">50000</span></code> is the window size around the gene body to consider for gene specificity scores.
If a SNP is within the gene window, it will be assigned the gene specificity score of that gene. If a SNP is within the window of multiple genes, the nearest gene will be used.</p>
<p><strong>Execution</strong>:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nv">LDScoreDir</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$WORKDIR</span><span class="s2">/</span><span class="nv">$SAMPLE_NAME</span><span class="s2">/generate_ldscore&quot;</span>
<span class="k">for</span><span class="w"> </span>CHROM<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="o">{</span><span class="m">1</span>..22<span class="o">}</span><span class="p">;</span><span class="w"> </span><span class="k">do</span>
<span class="w">    </span>GPS<span class="w"> </span>run_generate_ldscore<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--sample_name<span class="w"> </span><span class="nv">$SAMPLE_NAME</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--chrom<span class="w"> </span><span class="nv">$CHROM</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--ldscore_save_dir<span class="w"> </span><span class="nv">$LDScoreDir</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--mkscore_feather_file<span class="w"> </span><span class="nv">$MKSCORE_FEATHER_PATH</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--bfile_root<span class="w"> </span><span class="nv">$BFILE_ROOT</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--keep_snp_root<span class="w"> </span><span class="nv">$KEEP_SNP_ROOT</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--gtf_annotation_file<span class="w"> </span><span class="nv">$GTFFILE</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--gene_window_size<span class="w"> </span><span class="m">50000</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--spots_per_chunk<span class="w"> </span><span class="m">1000</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--ld_wind<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--ld_unit<span class="w"> </span><span class="s2">&quot;CM&quot;</span>
<span class="k">done</span>
</pre></div>
</div>
</div>
<input class="tab-input" id="tab-set--0-input--2" name="tab-set--0" type="radio"><label class="tab-label" for="tab-set--0-input--2">2. Use Enhancer-Gene Linking Only</label><div class="tab-content docutils">
<p>When a SNP mapped to multiple enhancers, the gene specificity score of the SNP will be set by the <code class="docutils literal notranslate"><span class="pre">--snp_multiple_enhancer_strategy</span></code> parameter. The default value is <code class="docutils literal notranslate"><span class="pre">max_mkscore</span></code>, which means the gene specificity score of the SNP will be set to the maximum gene specificity score of the enhancers that the SNP mapped to. Possible choices: max_mkscore, nearest_TSS</p>
<p>In this example we choose the brain tissue enhancer annotation file (<code class="docutils literal notranslate"><span class="pre">ENHANCER_ANNOTATION_FILE</span></code>).</p>
<p><strong>Execution</strong>:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nv">LDScoreDir</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$WORKDIR</span><span class="s2">/</span><span class="nv">$SAMPLE_NAME</span><span class="s2">/generate_ldscore&quot;</span>
<span class="k">for</span><span class="w"> </span>CHROM<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="o">{</span><span class="m">1</span>..22<span class="o">}</span><span class="p">;</span><span class="w"> </span><span class="k">do</span>
<span class="w">    </span>GPS<span class="w"> </span>run_generate_ldscore<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--sample_name<span class="w"> </span><span class="nv">$SAMPLE_NAME</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--chrom<span class="w"> </span><span class="nv">$CHROM</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--ldscore_save_dir<span class="w"> </span><span class="nv">$LDScoreDir</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--mkscore_feather_file<span class="w"> </span><span class="nv">$MKSCORE_FEATHER_PATH</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--bfile_root<span class="w"> </span><span class="nv">$BFILE_ROOT</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--keep_snp_root<span class="w"> </span><span class="nv">$KEEP_SNP_ROOT</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--gtf_annotation_file<span class="w"> </span><span class="nv">$GTFFILE</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--enhancer_annotation_file<span class="w"> </span><span class="nv">$ENHANCER_ANNOTATION_FILE</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--snp_multiple_enhancer_strategy<span class="w"> </span><span class="s1">&#39;max_mkscore&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--gene_window_enhancer_priority<span class="w"> </span><span class="s1">&#39;enhancer_only&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--spots_per_chunk<span class="w"> </span><span class="m">1000</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--ld_wind<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--ld_unit<span class="w"> </span><span class="s2">&quot;CM&quot;</span>
<span class="k">done</span>
</pre></div>
</div>
</div>
<input class="tab-input" id="tab-set--0-input--3" name="tab-set--0" type="radio"><label class="tab-label" for="tab-set--0-input--3">3. Use Both TSS and Enhancer-Gene Linking</label><div class="tab-content docutils">
<p>This will use both TSS and enhancer-gene linking to link SNPs to gene specificity.</p>
<p>In the scenario where a single nucleotide polymorphism (SNP) falls within the proximity window of both a gene body window and an enhancer linked to a different gene, the determination of the SNP’s gene specificity score is governed by the <code class="docutils literal notranslate"><span class="pre">--gene_window_enhancer_priority</span></code> parameter. Possible choices is <code class="docutils literal notranslate"><span class="pre">gene_window_first</span></code> or <code class="docutils literal notranslate"><span class="pre">enhancer_first</span></code>.</p>
<p><strong>Execution</strong>:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nv">LDScoreDir</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$WORKDIR</span><span class="s2">/</span><span class="nv">$SAMPLE_NAME</span><span class="s2">/generate_ldscore&quot;</span>
<span class="k">for</span><span class="w"> </span>CHROM<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="o">{</span><span class="m">1</span>..22<span class="o">}</span><span class="p">;</span><span class="w"> </span><span class="k">do</span>
<span class="w">    </span>GPS<span class="w"> </span>run_generate_ldscore<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--sample_name<span class="w"> </span><span class="nv">$SAMPLE_NAME</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--chrom<span class="w"> </span><span class="nv">$CHROM</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--ldscore_save_dir<span class="w"> </span><span class="nv">$LDScoreDir</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--mkscore_feather_file<span class="w"> </span><span class="nv">$MKSCORE_FEATHER_PATH</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--bfile_root<span class="w"> </span><span class="nv">$BFILE_ROOT</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--keep_snp_root<span class="w"> </span><span class="nv">$KEEP_SNP_ROOT</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--gtf_annotation_file<span class="w"> </span><span class="nv">$GTFFILE</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--gene_window_size<span class="w"> </span><span class="m">50000</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--enhancer_annotation_file<span class="w"> </span>ENHANCER_ANNOTATION_FILE<span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--snp_multiple_enhancer_strategy<span class="w"> </span><span class="s1">&#39;max_mkscore&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--gene_window_enhancer_priority<span class="w"> </span><span class="s1">&#39;gene_window_first&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--spots_per_chunk<span class="w"> </span><span class="m">1000</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--ld_wind<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="se">\</span>
<span class="w">        </span>--ld_unit<span class="w"> </span><span class="s2">&quot;CM&quot;</span>
<span class="k">done</span>
</pre></div>
</div>
</div>
</div>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>If you runout of memory in this step or the next step,
you can reduce the <code class="docutils literal notranslate"><span class="pre">--spots_per_chunk</span></code> parameter to a smaller value.</p>
<p>In general, 40GB memory is required when <code class="docutils literal notranslate"><span class="pre">--spots_per_chunk</span></code> is set to 1000.</p>
</div>
</section>
<section id="spatial-ldsc">
<h3>2.4 spatial_ldsc<a class="headerlink" href="#spatial-ldsc" title="Permalink to this heading">#</a></h3>
<p><strong>Objective</strong>: Identify spots associated with complex traits using spatial S-LDSC method.</p>
<p><strong>Input</strong>:</p>
<ul class="simple">
<li><p>Directory containing LD scores generated in the previous step (<code class="docutils literal notranslate"><span class="pre">LDScoreDir</span></code>).</p></li>
</ul>
<p><strong>Output</strong>:</p>
<ul class="simple">
<li><p>Spatial LDSC results saved in the specified directory (<code class="docutils literal notranslate"><span class="pre">LDSC_SAVE_DIR</span></code>) for each trait.</p></li>
</ul>
<div class="tab-set docutils">
<input checked="True" class="tab-input" id="tab-set--1-input--1" name="tab-set--1" type="radio"><label class="tab-label" for="tab-set--1-input--1">Run single trait</label><div class="tab-content docutils">
<p><strong>Execution</strong>:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nv">LDSC_DIR</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$WORKDIR</span><span class="s2">/</span><span class="nv">$SAMPLE_NAME</span><span class="s2">/spatial_ldsc&quot;</span>
<span class="nv">SUMSTATS_FILE</span><span class="o">=</span><span class="s2">&quot;example_data/GWAS/IQ_NG_2018.sumstats.gz&quot;</span>
<span class="nv">TRAIT_NAME</span><span class="o">=</span><span class="s2">&quot;IQ&quot;</span>

GPS<span class="w"> </span>run_spatial_ldsc<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--sumstats_file<span class="w"> </span><span class="nv">$SUMSTATS_FILE</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--trait_name<span class="w"> </span><span class="nv">$TRAIT_NAME</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--w_file<span class="w"> </span><span class="nv">$W_FILE</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--sample_name<span class="w"> </span><span class="nv">$SAMPLE_NAME</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--num_processes<span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--ldscore_input_dir<span class="w"> </span><span class="nv">$LDScoreDir</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--ldsc_save_dir<span class="w"> </span><span class="nv">$LDSC_DIR</span><span class="w"> </span><span class="se">\</span>
</pre></div>
</div>
</div>
<input class="tab-input" id="tab-set--1-input--2" name="tab-set--1" type="radio"><label class="tab-label" for="tab-set--1-input--2">Run multiple traits in batch</label><div class="tab-content docutils">
<p><strong>Execution</strong>:</p>
<p>You could run multiple traits in batch by providing a YAML file with the <code class="docutils literal notranslate"><span class="pre">--sumstats_config_file</span></code> parameter. An example of <code class="docutils literal notranslate"><span class="pre">sumstats_config_file</span></code> is like this:</p>
<div class="highlight-yaml notranslate"><div class="highlight"><pre><span></span><span class="nt">Height</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">example_data/GWAS/GIANT_EUR_Height_2022_Nature.sumstats.gz</span>
<span class="nt">IQ</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">example_data/GWAS/IQ_NG_2018.sumstats.gz</span>
<span class="nt">MCHC</span><span class="p">:</span><span class="w"> </span><span class="l l-Scalar l-Scalar-Plain">example_data/GWAS/BCX2_MCHC_EA_GWAMA.sumstats.gz</span>
</pre></div>
</div>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nv">LDSC_DIR</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$WORKDIR</span><span class="s2">/</span><span class="nv">$SAMPLE_NAME</span><span class="s2">/spatial_ldsc&quot;</span>
<span class="nv">SUMSTATS_CONFIG_FILE</span><span class="o">=</span><span class="s2">&quot;example_data/GWAS/gwas_config.yaml&quot;</span>
GPS<span class="w"> </span>run_spatial_ldsc<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--w_file<span class="w"> </span><span class="nv">$W_FILE</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--sample_name<span class="w"> </span><span class="nv">$SAMPLE_NAME</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--num_processes<span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--ldscore_input_dir<span class="w"> </span><span class="nv">$LDScoreDir</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--ldsc_save_dir<span class="w"> </span><span class="nv">$LDSC_DIR</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--sumstats_config_file<span class="w"> </span><span class="nv">$SUMSTATS_CONFIG_FILE</span>
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="post-processing">
<h2>3. Post-processing<a class="headerlink" href="#post-processing" title="Permalink to this heading">#</a></h2>
<section id="cauchy-combination-optional">
<h3>3.1 cauchy_combination (optional)<a class="headerlink" href="#cauchy-combination-optional" title="Permalink to this heading">#</a></h3>
<p><strong>Objective</strong>: Use the Cauchy Combination Test to aggregate P values of individual spots within specific spatial regions or cell types to evaluate the association of these regions with complex traits.</p>
<p><strong>Execution</strong>:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="nv">CAUCHY_SAVE_DIR</span><span class="o">=</span><span class="s2">&quot;</span><span class="nv">$WORKDIR</span><span class="s2">/</span><span class="nv">$SAMPLE_NAME</span><span class="s2">/cauchy_combination&quot;</span>
<span class="nv">TRAIT_NAME</span><span class="o">=</span><span class="s2">&quot;IQ&quot;</span>
GPS<span class="w"> </span>run_cauchy_combination<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--input_hdf5_path<span class="w"> </span><span class="nv">$HDF5_PATH</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--input_ldsc_dir<span class="w"> </span><span class="nv">$LDSC_DIR</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--sample_name<span class="w"> </span><span class="nv">$SAMPLE_NAME</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--output_cauchy_dir<span class="w"> </span><span class="nv">$CAUCHY_SAVE_DIR</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--trait_name<span class="w"> </span><span class="nv">$TRAIT_NAME</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--annotation<span class="w"> </span><span class="nv">$ANNOTATION</span>
</pre></div>
</div>
<p>You will get a csv file with the aggregated P values for each region or cell type to find out the most significant regions or cell types associated with the complex trait.</p>
<div class="tab-set docutils">
<input checked="True" class="tab-input" id="tab-set--2-input--1" name="tab-set--2" type="radio"><label class="tab-label" for="tab-set--2-input--1">Height</label><div class="tab-content docutils">
<div class="table-wrapper colwidths-auto docutils container">
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>p_cauchy</p></th>
<th class="head"><p>p_median</p></th>
<th class="head"><p>annotation</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>2.20e-06</p></td>
<td><p>6.72e-04</p></td>
<td><p>Layer1</p></td>
</tr>
<tr class="row-odd"><td><p>4.60e-02</p></td>
<td><p>2.80e-01</p></td>
<td><p>Layer2</p></td>
</tr>
<tr class="row-even"><td><p>8.64e-03</p></td>
<td><p>2.18e-01</p></td>
<td><p>Layer3</p></td>
</tr>
<tr class="row-odd"><td><p>5.75e-03</p></td>
<td><p>2.05e-01</p></td>
<td><p>Layer4</p></td>
</tr>
<tr class="row-even"><td><p>5.92e-04</p></td>
<td><p>1.34e-01</p></td>
<td><p>Layer5</p></td>
</tr>
<tr class="row-odd"><td><p>1.42e-02</p></td>
<td><p>1.34e-01</p></td>
<td><p>Layer6</p></td>
</tr>
<tr class="row-even"><td><p>1.95e-09</p></td>
<td><p>1.26e-04</p></td>
<td><p>WM</p></td>
</tr>
</tbody>
</table>
</div>
</div>
<input class="tab-input" id="tab-set--2-input--2" name="tab-set--2" type="radio"><label class="tab-label" for="tab-set--2-input--2">IQ</label><div class="tab-content docutils">
<div class="table-wrapper colwidths-auto docutils container">
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>p_cauchy</p></th>
<th class="head"><p>p_median</p></th>
<th class="head"><p>annotation</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>1.00e+00</p></td>
<td><p>9.91e-01</p></td>
<td><p>Layer1</p></td>
</tr>
<tr class="row-odd"><td><p>5.77e-15</p></td>
<td><p>3.00e-10</p></td>
<td><p>Layer2</p></td>
</tr>
<tr class="row-even"><td><p>2.09e-16</p></td>
<td><p>3.15e-07</p></td>
<td><p>Layer3</p></td>
</tr>
<tr class="row-odd"><td><p>6.77e-10</p></td>
<td><p>5.26e-06</p></td>
<td><p>Layer4</p></td>
</tr>
<tr class="row-even"><td><p>3.33e-15</p></td>
<td><p>3.73e-08</p></td>
<td><p>Layer5</p></td>
</tr>
<tr class="row-odd"><td><p>1.04e-12</p></td>
<td><p>1.70e-06</p></td>
<td><p>Layer6</p></td>
</tr>
<tr class="row-even"><td><p>1.00e+00</p></td>
<td><p>9.73e-01</p></td>
<td><p>WM</p></td>
</tr>
</tbody>
</table>
</div>
</div>
<input class="tab-input" id="tab-set--2-input--3" name="tab-set--2" type="radio"><label class="tab-label" for="tab-set--2-input--3">MCHC</label><div class="tab-content docutils">
<div class="table-wrapper colwidths-auto docutils container">
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>p_cauchy</p></th>
<th class="head"><p>p_median</p></th>
<th class="head"><p>annotation</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>2.14e-04</p></td>
<td><p>0.0641</p></td>
<td><p>Layer1</p></td>
</tr>
<tr class="row-odd"><td><p>0.0293</p></td>
<td><p>0.170</p></td>
<td><p>Layer2</p></td>
</tr>
<tr class="row-even"><td><p>0.00183</p></td>
<td><p>0.0422</p></td>
<td><p>Layer3</p></td>
</tr>
<tr class="row-odd"><td><p>0.00105</p></td>
<td><p>0.0133</p></td>
<td><p>Layer4</p></td>
</tr>
<tr class="row-even"><td><p>0.000244</p></td>
<td><p>0.0158</p></td>
<td><p>Layer5</p></td>
</tr>
<tr class="row-odd"><td><p>0.00064</p></td>
<td><p>0.0169</p></td>
<td><p>Layer6</p></td>
</tr>
<tr class="row-even"><td><p>3.54e-05</p></td>
<td><p>0.00696</p></td>
<td><p>WM</p></td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section>
<section id="visualization">
<h3>3.2 visualization<a class="headerlink" href="#visualization" title="Permalink to this heading">#</a></h3>
<p>You could use below command to visualize the spatial LDSC results. You will get a scatter plot with the -log10(p-value) of each spot.</p>
<p><strong>Output</strong></p>
<ul class="simple">
<li><p>A pdf file.</p></li>
<li><p>A html file which could be opened in a web browser to interactively explore the spatial LDSC results.</p></li>
</ul>
<p><strong>Execution</strong>:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>GPS<span class="w"> </span>run_visualize<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--input_hdf5_path<span class="w"> </span><span class="nv">$HDF5_PATH</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--input_ldsc_dir<span class="w"> </span><span class="nv">$LDSC_DIR</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--output_figure_dir<span class="w"> </span><span class="nv">$WORKDIR</span>/<span class="nv">$SAMPLE_NAME</span>/figures<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--sample_name<span class="w"> </span><span class="nv">$SAMPLE_NAME</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--trait_name<span class="w"> </span><span class="nv">$TRAIT_NAME</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--fig_title<span class="w"> </span><span class="nv">$TRAIT_NAME</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--annotation<span class="w"> </span><span class="nv">$ANNOTATION</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--point_size<span class="w"> </span><span class="m">7</span>
</pre></div>
</div>
</section>
</section>
</section>
<div class="tab-set docutils">
<input checked="True" class="tab-input" id="tab-set--3-input--1" name="tab-set--3" type="radio"><label class="tab-label" for="tab-set--3-input--1">Height</label><div class="tab-content docutils">
<div class="sphinx-charts docutils container" id="id1">
<div class="sphinx-charts-chart sphinx-charts-chart-id-0 sphinx-charts-chart-uri-_charts/charts/cortex/Cortex_151507_Height.json sphinx-charts-chart-dn-chart docutils container" id="sphinx-charts-chart-id-0">
</div>
<p class="sphinx-charts-hidden"></p>
<div class="sphinx-charts-placeholder sphinx-charts-placeholder-0 docutils container" id="id2">
<p><span class="caption-text">Loading…</span></p>
</div>
<p><span class="caption-text">Visualize the spatial heterogeneity of the height of the human cortex.</span></p>
</div>
</div>
<input class="tab-input" id="tab-set--3-input--2" name="tab-set--3" type="radio"><label class="tab-label" for="tab-set--3-input--2">IQ</label><div class="tab-content docutils">
<div class="sphinx-charts docutils container" id="id3">
<div class="sphinx-charts-chart sphinx-charts-chart-id-1 sphinx-charts-chart-uri-_charts/charts/cortex/Cortex_151507_IQ.json sphinx-charts-chart-dn-chart docutils container" id="sphinx-charts-chart-id-1">
</div>
<p class="sphinx-charts-hidden"></p>
<div class="sphinx-charts-placeholder sphinx-charts-placeholder-1 docutils container" id="id4">
<p><span class="caption-text">Loading…</span></p>
</div>
<p><span class="caption-text">Visualize the spatial heterogeneity of IQ in the human cortex.</span></p>
</div>
</div>
<input class="tab-input" id="tab-set--3-input--3" name="tab-set--3" type="radio"><label class="tab-label" for="tab-set--3-input--3">MCHC</label><div class="tab-content docutils">
<div class="sphinx-charts docutils container" id="id5">
<div class="sphinx-charts-chart sphinx-charts-chart-id-2 sphinx-charts-chart-uri-_charts/charts/cortex/Cortex_151507_MCHC.json sphinx-charts-chart-dn-chart docutils container" id="sphinx-charts-chart-id-2">
</div>
<p class="sphinx-charts-hidden"></p>
<div class="sphinx-charts-placeholder sphinx-charts-placeholder-2 docutils container" id="id6">
<p><span class="caption-text">Loading…</span></p>
</div>
<p><span class="caption-text">Visualize the spatial heterogeneity of MCHC in the human cortex.</span></p>
</div>
</div>
</div>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          
          
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2024, Liyang, Wenhao
            </div>
            Made with <a href="https://www.sphinx-doc.org/">Sphinx</a> and <a class="muted-link" href="https://pradyunsg.me">@pradyunsg</a>'s
            
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            On this page
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">Human Cortex Example</a><ul>
<li><a class="reference internal" href="#preparation">1. Preparation</a><ul>
<li><a class="reference internal" href="#download-running-dependencies">1.1 Download running dependencies</a></li>
<li><a class="reference internal" href="#download-example-data">1.2 Download example data</a></li>
</ul>
</li>
<li><a class="reference internal" href="#run-gps">2. Run GPS</a><ul>
<li><a class="reference internal" href="#find-latent-representations">2.1 find_latent_representations</a></li>
<li><a class="reference internal" href="#latent-to-gene">2.2 latent_to_gene</a></li>
<li><a class="reference internal" href="#generate-ldscore">2.3 generate_ldscore</a></li>
<li><a class="reference internal" href="#spatial-ldsc">2.4 spatial_ldsc</a></li>
</ul>
</li>
<li><a class="reference internal" href="#post-processing">3. Post-processing</a><ul>
<li><a class="reference internal" href="#cauchy-combination-optional">3.1 cauchy_combination (optional)</a></li>
<li><a class="reference internal" href="#visualization">3.2 visualization</a></li>
</ul>
</li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script src="_static/jquery.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/sphinx_highlight.js"></script>
    <script src="_static/scripts/furo.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-svg.js"></script>
    <script src="_static/sphinx_charts/plotly/plotly-2.11.1.min.js"></script>
    <script src="_static/sphinx_charts/charts.js"></script>
    <script src="_static/tabs.js"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script src="https://cdn.plot.ly/plotly-latest.min.js"></script>
    </body>
</html>