<!doctype html>
<html class="no-js" lang="en" data-content_root="">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />
<link rel="index" title="Index" href="genindex.html" /><link rel="search" title="Search" href="search.html" /><link rel="next" title="GPS Resources Download" href="data.html" /><link rel="prev" title="Mouse Embryo Example" href="mouse.html" />

    <!-- Generated with Sphinx 6.0.0 and Furo 2024.01.29 -->
        <title>Data Format - GPS 1.60 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/styles/furo.css?digest=369552022d0b975c8e74270ce6eabe0fb7978f24" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/tabs.css" />
    <link rel="stylesheet" type="text/css" href="_static/styles/furo-extensions.css?digest=30d1aed668e5c3a91c3e3bf6a60b675221979f0e" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="index.html"><div class="brand">GPS 1.60 documentation</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="index.html">
  
  
  <span class="sidebar-brand-text">GPS 1.60 documentation</span>
  
</a><form class="sidebar-search-container" method="get" action="search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="install.html">Installation guide</a></li>
<li class="toctree-l1 current has-children"><a class="reference internal" href="tutorials.html">GPS Tutorials</a><input checked="" class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle navigation of GPS Tutorials</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="mouse.html">Mouse Embryo Example</a></li>
<li class="toctree-l2 current current-page"><a class="current reference internal" href="#">Data Format</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="data.html">GPS Resources Download</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="api.html">GPS Command Line Alphabets</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" role="switch" type="checkbox"/><label for="toctree-checkbox-2"><div class="visually-hidden">Toggle navigation of GPS Command Line Alphabets</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="api/format_sumstats.html">format_sumstats (optional)</a></li>
<li class="toctree-l2"><a class="reference internal" href="api/find_latent_representations.html">Step 1: find_latent_representation</a></li>
<li class="toctree-l2"><a class="reference internal" href="api/latent_to_gene.html">Step 2: latent_to_gene</a></li>
<li class="toctree-l2"><a class="reference internal" href="api/generate_ldscore.html">Step 3: generate_ldscore</a></li>
<li class="toctree-l2"><a class="reference internal" href="api/spatial_ldsc.html">Step 4: spatial_ldsc</a></li>
<li class="toctree-l2"><a class="reference internal" href="api/cauchy_combination.html">Step 5: cauchy_combination (optional)</a></li>
<li class="toctree-l2"><a class="reference internal" href="api/visualization.html">Step 6: visualization (optional)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="release.html">Release notes</a></li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          
<div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main">
          <section id="data-format">
<span id="id1"></span><h1>Data Format<a class="headerlink" href="#data-format" title="Permalink to this heading">#</a></h1>
<section id="st-data">
<h2>ST Data<a class="headerlink" href="#st-data" title="Permalink to this heading">#</a></h2>
<p>The input ST data should be a h5ad file, which include, at least, gene expression matrix and spatial coordinates.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">scanpy</span> <span class="k">as</span> <span class="nn">sc</span>
<span class="n">HDF5_PATH</span><span class="o">=</span><span class="s2">&quot;example_data/ST/E16.5_E1S1.MOSTA.h5ad&quot;</span>
<span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">read_h5ad</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">HDF5_PATH</span><span class="si">}</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
<section id="gene-expression-matrix">
<h3>gene expression matrix<a class="headerlink" href="#gene-expression-matrix" title="Permalink to this heading">#</a></h3>
<p>The gene expression matrix should be stored in the <code class="docutils literal notranslate"><span class="pre">layers</span></code> attribute of the h5ad file.<br />
The parameter <code class="docutils literal notranslate"><span class="pre">Type</span></code> of the <code class="docutils literal notranslate"><span class="pre">find_latent_representations</span></code> specifies which data layer to use. If you set <code class="docutils literal notranslate"><span class="pre">Type</span></code> to <code class="docutils literal notranslate"><span class="pre">counts</span></code>, GPS will use the gene count matrix and automatically normalize it. If <code class="docutils literal notranslate"><span class="pre">Type</span></code> is set to another value, GPS will use the data provided in the corresponding layer without any normalization.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">adata</span><span class="o">.</span><span class="n">layers</span><span class="p">[</span><span class="s1">&#39;counts&#39;</span><span class="p">]</span>

<span class="go">&lt;121767x28204 sparse matrix of type &#39;&lt;class &#39;numpy.int64&#39;&gt;&#39;</span>
<span class="go">	with 211994186 stored elements in Compressed Sparse Row format&gt;</span>
</pre></div>
</div>
</section>
<section id="spatial-coordinates">
<h3>spatial coordinates<a class="headerlink" href="#spatial-coordinates" title="Permalink to this heading">#</a></h3>
<p>The spatial coordinates should be stored in the <code class="docutils literal notranslate"><span class="pre">obsm</span></code> attribute of the h5ad file with the key <code class="docutils literal notranslate"><span class="pre">spatial</span></code>.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">adata</span><span class="o">.</span><span class="n">obsm</span><span class="p">[</span><span class="s1">&#39;spatial&#39;</span><span class="p">]</span>

<span class="go">[[158.05955033 220.80121954]</span>
<span class="go"> [159.04435808 220.97486772]</span>
<span class="go"> [160.02916583 221.1485159 ]</span>
<span class="go"> ...</span>
<span class="go"> [112.83052085 730.69369339]</span>
<span class="go"> [113.8153286  730.86734157]</span>
<span class="go"> [114.80013636 731.04098975]]</span>
</pre></div>
</div>
</section>
<section id="spots-cells-annotation">
<h3>spots (cells) annotation<a class="headerlink" href="#spots-cells-annotation" title="Permalink to this heading">#</a></h3>
<p>Optionally, the h5ad file could contain annotations of the spots (cells) in the <code class="docutils literal notranslate"><span class="pre">obs</span></code> attribute.<br />
The parameter <code class="docutils literal notranslate"><span class="pre">annotation</span></code> of the <code class="docutils literal notranslate"><span class="pre">find_latent_representations</span></code> specifies the names of annotation columns in the <code class="docutils literal notranslate"><span class="pre">obs</span></code> attribute.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">adata</span><span class="o">.</span><span class="n">obs</span><span class="p">[</span><span class="s1">&#39;annotation&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span><span class="o">.</span><span class="n">head</span><span class="p">())</span>

<span class="go">Brain                17374</span>
<span class="go">Liver                14167</span>
<span class="go">Cavity               11287</span>
<span class="go">Muscle                9853</span>
<span class="go">Connective tissue     8803</span>
<span class="go">Name: annotation, dtype: int64</span>
</pre></div>
</div>
</section>
</section>
<section id="gwas-data">
<h2>GWAS Data<a class="headerlink" href="#gwas-data" title="Permalink to this heading">#</a></h2>
<p>The input GWAS data is in a gzip text file, which should, at least, include SNP (rs number), Z (Z-statistics), and N (sample size). Columns A1 and A2 represent the effect allele and non-effect allele, respectively, but they are not necessary. The column orders can be customized, but headers are keywords and will be used by GPS.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>zcat<span class="w"> </span>example_data/GWAS/IQ_NG_2018.sumstats.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>head<span class="w"> </span>-n<span class="w"> </span><span class="m">5</span>

SNP<span class="w">		</span>A1<span class="w">	</span>A2<span class="w">	</span>Z<span class="w">	</span>N
rs12184267<span class="w">	</span>T<span class="w">	</span>C<span class="w">	</span><span class="m">0</span>.916<span class="w">	</span><span class="m">225955</span>
rs12184277<span class="w">	</span>G<span class="w">	</span>A<span class="w">	</span><span class="m">0</span>.656<span class="w">	</span><span class="m">226215</span>
rs12184279<span class="w">	</span>A<span class="w">	</span>C<span class="w">	</span><span class="m">1</span>.050<span class="w">	</span><span class="m">226224</span>
rs116801199<span class="w">	</span>T<span class="w">	</span>G<span class="w">	</span><span class="m">0</span>.300<span class="w">	</span><span class="m">226626</span>
</pre></div>
</div>
<section id="tips-for-formatting-the-gwas-data">
<h3>Tips for formatting the GWAS data<a class="headerlink" href="#tips-for-formatting-the-gwas-data" title="Permalink to this heading">#</a></h3>
<p>You can convert the GWAS summary data into the required format using customized code. For ease of use, we’ve provided a command in GPS to do this. We’ll demonstrate how to use this command with the human height GWAS data as an example.</p>
<p>First, download the human height GWAS data and decompress it.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>wegt<span class="w"> </span>https://portals.broadinstitute.org/collaboration/giant/images/4/4e/GIANT_HEIGHT_YENGO_2022_GWAS_SUMMARY_STATS_ALL.gz
gzip<span class="w"> </span>-d<span class="w"> </span>GIANT_HEIGHT_YENGO_2022_GWAS_SUMMARY_STATS_ALL.gz
</pre></div>
</div>
<p>The first 5 lines of the file should look like this:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>head<span class="w"> </span>-n<span class="w"> </span><span class="m">5</span><span class="w"> </span>GIANT_HEIGHT_YENGO_2022_GWAS_SUMMARY_STATS_ALL

SNPID<span class="w">	</span>RSID<span class="w">	</span>CHR<span class="w">	</span>POS<span class="w">	</span>EFFECT_ALLELE<span class="w">	</span>OTHER_ALLELE<span class="w">	</span>EFFECT_ALLELE_FREQ<span class="w">	</span>BETA<span class="w">	</span>SE<span class="w">	</span>P<span class="w">	</span>N
<span class="m">1</span>:566875:C:T<span class="w">	</span>rs2185539<span class="w">	</span><span class="m">1</span><span class="w">	</span><span class="m">566875</span><span class="w">	</span>T<span class="w">	</span>C<span class="w">	</span><span class="m">0</span>.00413<span class="w">	</span><span class="m">0</span>.00762959<span class="w">	</span><span class="m">0</span>.0197<span class="w">	</span><span class="m">0</span>.699009<span class="w">	</span><span class="m">607057</span>
<span class="m">1</span>:728951:C:T<span class="w">	</span>rs11240767<span class="w">	</span><span class="m">1</span><span class="w">	</span><span class="m">728951</span><span class="w">	</span>T<span class="w">	</span>C<span class="w">	</span><span class="m">0</span>.0438<span class="w">	</span><span class="m">0</span>.0206708<span class="w">	</span><span class="m">0</span>.00956<span class="w">	</span><span class="m">0</span>.0306735<span class="w">	</span><span class="m">241218</span>
<span class="m">1</span>:734462:A:G<span class="w">	</span>rs12564807<span class="w">	</span><span class="m">1</span><span class="w">	</span><span class="m">734462</span><span class="w">	</span>A<span class="w">	</span>G<span class="w">	</span><span class="m">0</span>.896<span class="w">	</span><span class="m">0</span>.00426816<span class="w">	</span><span class="m">0</span>.0110<span class="w">	</span><span class="m">0</span>.697777<span class="w">	</span><span class="m">117377</span>
<span class="m">1</span>:752721:A:G<span class="w">	</span>rs3131972<span class="w">	</span><span class="m">1</span><span class="w">	</span><span class="m">752721</span><span class="w">	</span>G<span class="w">	</span>A<span class="w">	</span><span class="m">0</span>.786<span class="w">	</span><span class="m">0</span>.0000413581<span class="w">	</span><span class="m">0</span>.00205<span class="w">	</span><span class="m">0</span>.983898<span class="w">	</span><span class="m">1009515</span>
</pre></div>
</div>
<p>To convert the summary statistics, use the commands:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>GPS<span class="w"> </span>format_sumstats<span class="w"> </span><span class="se">\</span>
--sumstats<span class="w"> </span>GIANT_HEIGHT_YENGO_2022_GWAS_SUMMARY_STATS_ALL<span class="w"> </span>
--out<span class="w"> </span>HEIGHT
</pre></div>
</div>
<p>you will get HEIGHT.sumstats.gz, which should look like this:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>zcat<span class="w"> </span>HEIGHT.sumstats.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>head<span class="w"> </span>-n<span class="w"> </span><span class="m">5</span>

SNP<span class="w">		</span>A1<span class="w">	</span>A2<span class="w">	</span>Z<span class="w">	</span>N<span class="w">	</span>
rs3131969<span class="w">	</span>G<span class="w">	</span>A<span class="w">	</span><span class="m">0</span>.328<span class="w">	</span><span class="m">1494218</span>.000<span class="w">	</span>
rs3131967<span class="w">	</span>C<span class="w">	</span>T<span class="w">	</span><span class="m">0</span>.386<span class="w">	</span><span class="m">1488150</span>.000<span class="w">	</span>
rs12562034<span class="w">	</span>A<span class="w">	</span>G<span class="w">	</span><span class="m">1</span>.714<span class="w">	</span><span class="m">1554976</span>.000<span class="w">	</span>
rs4040617<span class="w">	</span>G<span class="w">	</span>A<span class="w">	</span>-0.463<span class="w">	</span><span class="m">1602016</span>.000<span class="w">	</span>
</pre></div>
</div>
<p>In most cases, this command can automatically recognize different column names of GWAS data, but you can also specify the column names to ensure correctness.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>GPS<span class="w"> </span>format_sumstats<span class="w"> </span><span class="se">\</span>
--sumstats<span class="w"> </span>GIANT_HEIGHT_YENGO_2022_GWAS_SUMMARY_STATS_ALL<span class="w"> </span>
--out<span class="w"> </span>HEIGHT<span class="w"> </span><span class="se">\</span>
--a1<span class="w"> </span>EFFECT_ALLELE<span class="w"> </span><span class="se">\</span>
--a2<span class="w"> </span>OTHER_ALLELE<span class="w"> </span><span class="se">\</span>
--p<span class="w"> </span>P<span class="w"> </span><span class="se">\</span>
--frq<span class="w"> </span>EFFECT_ALLELE_FREQ<span class="w"> </span><span class="se">\</span>
--snp<span class="w"> </span>RSID<span class="w"> </span><span class="se">\</span>
--beta<span class="w"> </span>BETA<span class="w"> </span><span class="se">\</span>
--se<span class="w"> </span>SE<span class="w"> </span><span class="se">\</span>
--n<span class="w"> </span>N
</pre></div>
</div>
<p>For more usage options, please see:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>GPS<span class="w"> </span>format_sumstats<span class="w"> </span>-h
</pre></div>
</div>
</section>
</section>
</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          <a class="next-page" href="data.html">
              <div class="page-info">
                <div class="context">
                  <span>Next</span>
                </div>
                <div class="title">GPS Resources Download</div>
              </div>
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
            </a>
          <a class="prev-page" href="mouse.html">
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
              <div class="page-info">
                <div class="context">
                  <span>Previous</span>
                </div>
                
                <div class="title">Mouse Embryo Example</div>
                
              </div>
            </a>
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2024, Liyang, Wenhao
            </div>
            Made with <a href="https://www.sphinx-doc.org/">Sphinx</a> and <a class="muted-link" href="https://pradyunsg.me">@pradyunsg</a>'s
            
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            On this page
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">Data Format</a><ul>
<li><a class="reference internal" href="#st-data">ST Data</a><ul>
<li><a class="reference internal" href="#gene-expression-matrix">gene expression matrix</a></li>
<li><a class="reference internal" href="#spatial-coordinates">spatial coordinates</a></li>
<li><a class="reference internal" href="#spots-cells-annotation">spots (cells) annotation</a></li>
</ul>
</li>
<li><a class="reference internal" href="#gwas-data">GWAS Data</a><ul>
<li><a class="reference internal" href="#tips-for-formatting-the-gwas-data">Tips for formatting the GWAS data</a></li>
</ul>
</li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script src="_static/jquery.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/sphinx_highlight.js"></script>
    <script src="_static/scripts/furo.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-svg.js"></script>
    <script src="_static/tabs.js"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script src="https://cdn.plot.ly/plotly-latest.min.js"></script>
    </body>
</html>